### PR TITLE
Don't 500 on a bad form_id

### PR DESCRIPTION
**Pull request by esoergel**
http://manage.dimagi.com/default.asp?233275
:blowfish:
@benrudolph @sravfeyn

### CODE REVIEW

File: corehq/ex-submodules/couchforms/models.py [hunk 1]
```diff
@@ -135,7 +136,13 @@ def get(cls, docid, rev=None, db=None, dynamic_properties=True):
...
                 if doc['doc_type'] in doc_types():
                     return doc_types()[doc['doc_type']].wrap(doc)
-                return XFormInstance.wrap(doc)
+                try:
+                    return XFormInstance.wrap(doc)
+                except WrappingAttributeError:
+                    raise ResourceNotFound(
+                        "The doc with _id {} and doc_type {} can't be wrapped "
+                        "as an XFormInstance".format(docid, doc['doc_type'])
+                    )
```
Comment (esoergel) [inline corehq/ex-submodules/couchforms/models.py]: This is the actual, functional change in the PR.

File: corehq/ex-submodules/couchforms/models.py [hunk 2]
```diff
@@ -322,10 +322,6 @@ def unarchive(self, user_id=None):
         XFormInstance.save(self)  # subclasses explicitly set the doc type so force regular save
         xform_unarchived.send(sender="couchforms", xform=self)
 
-    @property
-    def is_archived(self):
-        return self.doc_type == "XFormArchived"
```
Comment (esoergel) [inline corehq/ex-submodules/couchforms/models.py]: This method is duplicated exactly in the same class.

File: corehq/ex-submodules/couchforms/models.py [hunk 3]
```diff
@@ -5,33 +5,33 @@
 import hashlib
```
Comment (benrudolph) [inline corehq/ex-submodules/couchforms/models.py]: some good stuff in this commit

**Comment by benrudolph** [comment]
✅ on 👍

File: corehq/ex-submodules/couchforms/models.py (hunk 3 above)
Comment (esoergel) [inline corehq/ex-submodules/couchforms/models.py]: :muscle:

**Comment by esoergel** [comment]
:white_check_mark:

**Pull request merged**